src/lib/scripts/TBAUpdateTM.ts

```diff
@@ -0,0 +1,108 @@
+import { API_KEY } from "$env/static/private"
+import { getEventKey } from "./dbUtil"
+import { warn } from "@/consoleUtils"
+import { Endgame } from "@prisma/client"
+import { prisma } from "$lib/prisma"
+
+export async function TBAUpdateMatchData(match_key: string) {
+    const event_key = await getEventKey()
+    const verbose_match_key = event_key + "_" + match_key
+
+    const res = await fetch(
+        `https://www.thebluealliance.com/api/v3/match/${verbose_match_key}`,
+        {
+            method: "GET",
+            headers: {
+                "If-None-Match": "ETag",
+                "Accept": "application/json",
+                "X-TBA-Auth-Key": API_KEY,
+            },
+        }
+    )
+
+    if (!res.ok) {
+        warn(`TBA /match/${verbose_match_key} request failed: ${res.status}`)
+        return
+    }
+
+    const match = await res.json()
+    const red: number[] = (
+        match["alliances"]["red"]["team_keys"] as string[]
+    ).map(team => Number.parseInt(team.slice(3)))
+    const blue: number[] = (
+        match["alliances"]["blue"]["team_keys"] as string[]
+    ).map(team => Number.parseInt(team.slice(3)))
+    const scores = match["score_breakdown"]
+    const recorded_endgames = await prisma.teamMatch.findMany({
+        where: {
+            match_key: {
+                equals: verbose_match_key,
+            },
+        },
+        select: {
+            endgame: true,
+            team_key: true,
+        },
+    })
+
+    for (let i = 1; i <= 3; i++) {
+        const red_endgame = determineEndgame(
+            scores["red"][`endgameRobot${i}`],
+            recorded_endgames.find(match => match.team_key === red[i - 1])
+                ?.endgame ?? Endgame.None
+        )
+
+        const blue_endgame = determineEndgame(
+            scores["blue"][`endgameRobot${i}`],
+            recorded_endgames.find(match => match.team_key === blue[i - 1])
+                ?.endgame ?? Endgame.None
+        )
```

Comment:
There's _probably_ a better way of doing this, but it's non-obvious so your solution is fine :3